src/search/pdbs/utils.cc

```diff
@@ -28,6 +38,36 @@ int compute_total_pdb_size(
     return size;
 }
 
+vector<FactPair> get_goals_in_random_order(
+    const TaskProxy &task_proxy, const shared_ptr<utils::RandomNumberGenerator> &rng) {
+    vector<FactPair> goals;
+    for (FactProxy goal : task_proxy.get_goals()) {
+        goals.push_back(goal.get_pair());
+    }
+    rng->shuffle(goals);
+    return goals;
+}
+
+vector<int> get_non_goal_variables(const TaskProxy &task_proxy) {
+    GoalsProxy goals = task_proxy.get_goals();
+    int num_vars = task_proxy.get_variables().size();
+    vector<int> non_goal_variables;
+    non_goal_variables.reserve(num_vars - goals.size());
+    for (int var_id = 0; var_id < num_vars; ++var_id) {
+        bool is_goal_var = false;
+        for (FactProxy goal : goals) {
+            if (var_id == goal.get_variable().get_id()) {
+                is_goal_var = true;
+                break;
+            }
+        }
+        if (!is_goal_var) {
+            non_goal_variables.push_back(var_id);
+        }
+    }
+    return non_goal_variables;
+}
```

Comment:
I think it's simplest to create a `vector<bool>(num_vars, false)`, set the goal bits, then extract the info into a `vector<int>`.

Reply:
Yes, this would be much better than the current O(n^2) algorithm (which almost always becomes a problem eventually). Alternatively, something like set_difference can be used because we should be guaranteed that the variables occur in sorted order. (And even if that were not the case, sorting them and then using set_difference would be much more efficient.)